File: Data Structures & Algorithms/duplicate-integer/submission-2.cpp
class Solution {
public:
    bool hasDuplicate(vector<int>& nums) {
     int dup=0;
     sort(nums.begin(),nums.end());
     for(int i=1;i<nums.size();i++){
        if(nums[i]==nums[i-1]){
            dup=1;
            break;
        }
     }
     if(dup==1)return true;
     else return false;
    }
};

File: Data Structures & Algorithms/top-k-elements-in-list/submission-2.java
class Solution {
    public int[] topKFrequent(int[] nums, int k) {
       Map<Integer,Integer>mp=new HashMap<>();
       for(Integer it:nums){
        mp.put(it,mp.getOrDefault(it,0)+1);
       } 
       PriorityQueue<Map.Entry<Integer,Integer>>pq=new PriorityQueue<>((a,b)->a.getValue()-b.getValue());
       for(Map.Entry<Integer,Integer>entry:mp.entrySet()){
           pq.offer(entry);
           if(pq.size()>k){
              pq.poll();
           }
       }
       int []ans=new int[k];
       int i=0;
       while(!pq.isEmpty()){
            ans[i++]=pq.poll().getKey();
            
       }
       return ans;
    }
}
